File: Data Structures & Algorithms/top-k-elements-in-list/submission-6.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer, Integer> map=new HashMap<>();
        int[] ans=new int[k];
        List<Integer>[] temp=new List[nums.length+1];
        PriorityQueue<int[]> pq=new PriorityQueue<>((a,b)->{
            if (a[0]!=b[0]){
                return a[0]-b[0];
            }
            return a[1]-b[1];
        });
        for(int num: nums){
            map.merge(num,1,Integer::sum);
        }
        for(Map.Entry<Integer, Integer> e: map.entrySet()){
            int[] t={e.getValue(),e.getKey()};
            pq.offer(t);
        }
        while (pq.size()>k){
            pq.poll();
        }
        for(int i=0;i<k;i++){
            ans[i]=pq.poll()[1];
        }
        return ans;

    }
}

File: Data Structures & Algorithms/remove-element/submission-6.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        if len(nums)==0:
            return 0
        while len(nums)>0 and nums[-1]==val:
            nums.pop()
        i=0
        while i<len(nums):
            num=nums[i]
            if num==val:
                nums[i],nums[-1]=nums[-1],nums[i]
                nums.pop()
            else:
                i+=1
        if not nums:
            return 0
        return len(nums)